python/ray/tune/registry.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from types import FunctionType
+
+import ray
+from ray.local_scheduler import ObjectID
+from ray.tune.trainable import Trainable, wrap_function
+
+TRAINABLE_CLASS = "trainable_class"
+ENV_CREATOR = "env_creator"
+KNOWN_CATEGORIES = [TRAINABLE_CLASS, ENV_CREATOR]
+
+
+def register_trainable(name, trainable):
+    """Register a trainable function or class.
+
+    Args:
+        name (str): Name to register.
+        trainable (obj): Function or tune.Trainable object. Functions must
+            take (config, status_reporter) as arguments and will be
+            automatically converted into a trainable obj during registration.
+    """
+
+    if isinstance(trainable, FunctionType):
+        trainable = wrap_function(trainable)
+    else:
+        assert issubclass(trainable, Trainable), trainable
+    _register(TRAINABLE_CLASS, name, trainable)
+
+
+def register_env(name, env_creator):
+    """Register a custom environment for use with RLlib.
+
+    Args:
+        name (str): Name to register.
+        env_creator (obj): Function that creates an env.
+    """
+
+    assert issubclass(env_creator, FunctionType)
+    _register(ENV_CREATOR, name, env_creator)
+
+
+def _register(category, key, value):
```

Comment:
This is fine - I'm curious why you made this design choice to abstract this function out

Reply:
Removed